cdap-ui/app/cdap/components/DataPrep/store/DataPrepActionCreator.js

```diff
@@ -296,3 +298,271 @@ export function setVisualizationState(state) {
     },
   });
 }
+
+export function setError(error, prefix) {
+  const status = error.statusCode;
+  const detail = error.message || error.response.message || 'Unknown error';
```

Comment:
This could also be a 404 in which case the message will be in `error.response`. I understand this is not formalized yet but can we add a check for error.response to be a string and if so use that?

Reply:
Done.